server/src/main/java/io/crate/metadata/information/InformationColumnsTableInfo.java

```diff
@@ -81,7 +82,18 @@
                     return null;
                 }
             })
-            .add("character_maximum_length", INTEGER, ignored -> null)
+            .add("character_maximum_length", INTEGER, r -> {
+                if (DataTypes.isSameType(r.info.valueType(), STRING)) {
```

Comment:
Minor: this could be simplified. I would prefer, however this is up to you.
```
if (DataTypes.isSameType(r.info.valueType(), STRING)) {
	var stringType = ((StringType) r.info.valueType());
	if (!stringType.unbound()) {
	  return stringType.lengthLimit();
	}
}
return null; 
```

Reply:
not sure, I find a bit more explicit and consistent if-else flow a bit easier to read, but no strong opinion.